pkg/controller/instance/delta/delta.go

```diff
@@ -113,7 +113,17 @@ func walkCompare(desired, observed interface{}, path string, differences *[]Diff
 		}
 		walkMap(d, e, path, differences)
 
+	case nil:
+		// Special case: treat empty array and nil as equivalent
+		if isEmptyArrayOrNil(desired) && isEmptyArrayOrNil(observed) {
```

Comment:
Handle for map ?

Reply:
Yeah IMO we should probably handle for map too?

Reply:
done.